core/src/main/kotlin/in/specmatic/core/pattern/RFC3339.kt

```diff
@@ -4,18 +4,31 @@ import java.time.LocalDateTime
 import java.time.ZoneId
 import java.time.ZonedDateTime
 import java.time.format.DateTimeFormatter
+import java.util.regex.Pattern
 
 private const val DATETIME_FORMAT = "yyyy-MM-dd'T'HH:mm:ssXXX"
 private const val DATE_FORMAT = "yyyy-MM-dd"
 
+private val RFC3339_PATTERN: Pattern = Pattern.compile (
+    "^(\\d{4})-(\\d{2})-(\\d{2})" // yyyy-MM-dd
+            + "([Tt](\\d{2}):(\\d{2}):(\\d{2})(\\.\\d+)?)?" // 'T'HH:mm:ss.milliseconds
+            + "([Zz]|([+-])(\\d{2}):(\\d{2}))?" // 'Z' or time zone shift HH:mm following '+' or '-'
+)
+
 class RFC3339 {
     companion object{
-        val dateTimeFormatter: DateTimeFormatter = DateTimeFormatter.ofPattern(DATETIME_FORMAT)
         val dateFormatter: DateTimeFormatter = DateTimeFormatter.ofPattern(DATE_FORMAT)
 
+        fun parse(dateTime: String) {
+            val matcher = RFC3339_PATTERN.matcher(dateTime)
+            if(!matcher.matches()) {
+                throw ContractException("Error while parsing the dateTime: $dateTime")
```

Comment:
Is there any match error message that can be appended to this Exception? Will help to say why the parse failed.

Reply:
I did not find one. Seems like there isn't one.